staker-rewards/lib/fraud-detector.js

```diff
@@ -0,0 +1,47 @@
+import clc from "cli-color"
+
+export default class FraudDetector {
+  constructor(context, tokenStaking, factoryDeploymentBlock) {
+    this.context = context
+
+    this.tokenStaking = tokenStaking
+
+    this.factoryDeploymentBlock = factoryDeploymentBlock
+  }
+
+  static async initialize(context) {
+    const { TokenStaking, factoryDeploymentBlock } = context.contracts
+
+    const tokenStaking = await TokenStaking.deployed()
+
+    return new FraudDetector(context, tokenStaking, factoryDeploymentBlock)
+  }
+
+  async isOperatorFraudulent(operator) {
+    // Get all slashing events for the operator.
+    const events = await this.tokenStaking.getPastEvents("TokensSlashed", {
+      fromBlock: this.factoryDeploymentBlock,
+      toBlock: "latest",
+      filter: { operator: operator },
+    })
+
+    // At this moment token slashing can only be originated from tBTC application
+    // on fraud detection. Random Beacon does not use slashing but seizing. We
+    // assume that any slashing event is related to a fraud detection for ECDSA
+    // Keep.
+    // This section has to be revisited in case of implementing additional usage
+    // of slashing function.
+    if (events.length > 0) {
+      console.log(
+        clc.yellow(
+          `Found slashing event for operator [${operator}];` +
+            ` please double check correctness of a calculation result`
```

Comment:
I would drop `please double check correctness of a calculation result` part to avoid confusion of other parties running this script. IMO it gives an impression something is potentially wrong. Calculation results should be inspected anyway, and this situation is not exceptional more than e.g. not meeting the SLA or withdrawing bonds.